### PR TITLE
Count, exists & upsert

### PR DESCRIPTION
**Pull request by artkay**
Adding some stuff here.

### CODE REVIEW

File: Sources/Query.swift [hunk 1]
```diff
@@ -53,6 +69,21 @@ class Query {
...
 
+	func exists() {
+		guard let table = self.table else {
+			return
+		}
+
+		return Query.driver.exists(table: table, filters: self.filters)
+	}
+
+	func count() -> [[String: String]] {
```
Comment (tannernelson) [inline Sources/Query.swift]: sig should be `func count() -> Int`

File: Sources/Query.swift [hunk 2]
```diff
@@ -53,6 +69,21 @@ class Query {
...
+		guard let table = self.table else {
+			return []
```
Comment (tannernelson) [inline Sources/Query.swift]: return 0

File: Sources/Query.swift [hunk 3]
```diff
@@ -53,6 +69,21 @@ class Query {
 		Query.driver.delete(table: table, filters: self.filters)
 	}
 
+	func exists() {
```
Comment (tannernelson) [inline Sources/Query.swift]: `func exists() -> Bool {`

File: Sources/Query.swift (hunk 1 above)
Comment (artkay) [inline Sources/Query.swift]: Oh it was like that, but then I hit CMD-Z and forgot about it.

File: Sources/Query.swift [hunk 4]
```diff
@@ -53,6 +69,21 @@ class Query {
...
+		guard let table = self.table else {
+			return
```
Comment (tannernelson) [inline Sources/Query.swift]: return false

File: Sources/Query.swift [hunk 5]
```diff
@@ -37,6 +37,22 @@ class Query {
...
 
+	func upsert(data: [[String: String]]) {
+		guard let table = self.table else {
+			return
+		}
+
+		Query.driver.upsert(table: table, items: data)
+	}
+
+		func upsert(data: [String: String]) {
```
Comment (tannernelson) [inline Sources/Query.swift]: Indentation

**Comment by tannernelson** [comment]
Count and exists are great. We should add these methods to the `Driver` protocol so they are required to be implemented by any drivers. That will force the `PrintDriver` to implement it.

**Comment by artkay** [comment]
Just added it

**Comment by tannernelson** [comment]
:+1:

**Pull request merged**